File: apps/api/app/schemas/schedule.py
from datetime import datetime
from pydantic import BaseModel, Field


class AppointmentCreate(BaseModel):
    patient_id: str
    provider_id: str
    start_time: datetime
    end_time: datetime
    type: str = "office_visit"
    notes: str | None = None


class AppointmentUpdate(BaseModel):
    start_time: datetime | None = None
    end_time: datetime | None = None
    type: str | None = None
    status: str | None = None
    notes: str | None = None


class AppointmentOut(BaseModel):
    id: str
    patient_id: str
    patient_name: str | None = None
    provider_id: str
    provider_name: str | None = None
    start_time: str
    end_time: str
    type: str
    status: str
    notes: str | None
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class AppointmentListOut(BaseModel):
    data: list[AppointmentOut]
    total: int


class TodayQueueItemOut(BaseModel):
    appointment: AppointmentOut
    checkout_readiness: str
    blockers: list[str]
    documents_needing_review: int
    open_tasks: int
    urgent_tasks: int
    unsigned_encounters: int


class TodayQueueOut(BaseModel):
    data: list[TodayQueueItemOut]
    total: int
    checked_in: int
    blocked: int


class AvailabilityCreate(BaseModel):
    provider_id: str
    day_of_week: int = Field(ge=0, le=6)
    start_time: str = Field(pattern=r"^\d{2}:\d{2}$")
    end_time: str = Field(pattern=r"^\d{2}:\d{2}$")


class AvailabilityOut(BaseModel):
    id: str
    provider_id: str
    day_of_week: int
    start_time: str
    end_time: str

    model_config = {"from_attributes": True}


class AppointmentReminderOut(BaseModel):
    appointment_id: str
    queued: int
    event_ids: list[str]


class AppointmentConflictCheckOut(BaseModel):
    provider_id: str
    start_time: str
    end_time: str
    has_conflict: bool
    in_availability: bool
    warnings: list[str]

File: packages/shared/src/types/schedule.ts
import type { UUID, AppointmentStatus } from './common';

export interface Appointment {
  id: UUID;
  patient_id: UUID;
  patient_name: string;
  provider_id: UUID;
  provider_name: string;
  start_time: string;
  end_time: string;
  type: string;
  status: AppointmentStatus;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface TodayQueueItem {
  appointment: Appointment;
  checkout_readiness: 'ready' | 'blocked';
  blockers: string[];
  documents_needing_review: number;
  open_tasks: number;
  urgent_tasks: number;
  unsigned_encounters: number;
  medications_needing_review: number;
  labs_needing_review: number;
  care_plan_blockers: number;
}

export interface TodayQueue {
  data: TodayQueueItem[];
  total: number;
  checked_in: number;
  blocked: number;
}

export interface ProviderAvailability {
  id: UUID;
  provider_id: UUID;
  day_of_week: number;
  start_time: string;
  end_time: string;
}

export interface AppointmentReminderQueue {
  appointment_id: UUID;
  queued: number;
  event_ids: UUID[];
}

export interface AppointmentConflictCheck {
  provider_id: UUID;
  start_time: string;
  end_time: string;
  has_conflict: boolean;
  in_availability: boolean;
  warnings: string[];
  suggested_slots: { start_time: string; end_time: string }[];
}
